Feature

djkowa opened the pull request:
about page defined

Commit:
android x changes

Commit:
about page inserted

Commit:
about page

matejthetree commented on lib/ui_pages/about_page.dart:
```diff
@@ -0,0 +1,136 @@
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
```
Ne treba material app tu

matejthetree commented on lib/ui_pages/about_page.dart:
```diff
@@ -0,0 +1,136 @@
+      home: Scaffold(
+       endDrawer: CustomDrawer(),
+        appBar: AppBar(
+          automaticallyImplyLeading: false,
```
Ovdje ostaje leading back arrow. Actions je samo menu button koji treba biti widget za sebe. Pa ga samo stavimo. Uvijek poziva scaffold of

matejthetree commented on lib/ui_pages/about_page.dart:
```diff
@@ -0,0 +1,136 @@
+                  )
+                  ),
+                ),
+                Text('About Page'),
```
Title je samo Text. Da si htio button prije titlea staviti ručno, isllo bi `leading: IconButton...`

matejthetree commented on lib/ui_pages/home.dart:
```diff
@@ -0,0 +1,36 @@
+       endDrawer: CustomDrawer(),
+        appBar: AppBar(
+          automaticallyImplyLeading: false,
+          title: Center(child: Text('Mokuteki.io Playgrounds')),
```
Ne treba center. Ima appbar centerTitle property

matejthetree commented on lib/ui_pages/home.dart:
```diff
@@ -0,0 +1,36 @@
+        appBar: AppBar(
+          automaticallyImplyLeading: false,
+          title: Center(child: Text('Mokuteki.io Playgrounds')),
+          backgroundColor: Colors.pink,
```
Stavi u material app primary color kao pink pa ovo nece trebati.

matejthetree commented on lib/ui_pages/home.dart:
```diff
@@ -0,0 +1,36 @@
+          title: Center(child: Text('Mokuteki.io Playgrounds')),
+          backgroundColor: Colors.pink,
+          actions: <Widget>[
+
```
Ata nebi tu u action trebao ici menuButton

matejthetree commented on lib/widgets/drawer.dart:
```diff
@@ -0,0 +1,33 @@
+            child: FlatButton(
+              child: Text('About',style: TextStyle(fontSize: 20,fontStyle: FontStyle.italic) ),
+              onPressed: () =>
+                  Navigator.push(context, MaterialPageRoute(
```
Treba pushat routu RouteName.about, umjesto da je tu radi.

matejthetree commented on lib/widgets/menu_button.dart:
```diff
@@ -0,0 +1,18 @@
+      icon: Icon(
+        Icons.menu,
+      ),
+      onPressed: () => Scaffold.of(context).openDrawer(),
```
Open end drawer

lea88pu replied:
@matejthetree Probali smo staviti `.openEndDrawer(),` ali kada se pozove `MenuButton()` u actions, ne otvara nista. Tek kada u `Scaffold` dodamo `endDrawer:CustomDrawer(),` ali onda on zapravo niti ne koristi taj `MenuButton()`

matejthetree replied:
kako ne koristi, ne razumijem. 
u actions u scaffoldu ces dodati [MenuDrawer()]

lea88pu replied:
@matejthetree Pa kada maknem iz Scaffolda `endDrawer:`,  MenuButton() ne otvara ladicu.
Ovako radi:
`class _AboutPageState extends State<AboutPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      endDrawer: CustomDrawer(),
      appBar: AppBar(
        title: Text('About Page'),
      ),` 
A ovako ne radi:
`class _AboutPageState extends State<AboutPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(actions: <Widget>[
        MenuButton(),
      ],
        title: Text('About Page'),
      ),`

matejthetree replied:
tocno. svaki scaffold treba imati end drawer. inace ga nema :D

matejthetree replied:
menu button ces dodati samo u scaffold koji ima end drawer.

lea88pu replied:
Ali zasto da ga stavim kad radi i bez njega?

matejthetree commented:
Niste mi pročitali code comments. 

Na svaki pročitani treba ici :thumbsup: ili komentar

matejthetree commented on .gitignore:
```diff
@@ -24,50 +24,14 @@
 **/doc/api/
```
Test komentar

djkowa commented on .gitignore:
```diff
@@ -24,50 +24,14 @@
 **/doc/api/
 .dart_tool/
 .flutter-plugins
+.flutter-plugins-dependencies
 .packages
```
test comment

matejthetree commented on .gitignore:
```diff
@@ -24,50 +24,14 @@
 .dart_tool/
 .flutter-plugins
+.flutter-plugins-dependencies
 .packages
 .pub-cache/
 .pub/
 /build/
 
-# Android related
-**/android/**/gradle-wrapper.jar
```
zasto ne radi

matejthetree commented:
tocno.

u svaki scaffold moras dodati end drawer

On Fri, Dec 13, 2019 at 6:24 PM lea88pu <notifications@github.com> wrote:

> *@lea88pu* commented on this pull request.
> ------------------------------
>
> In lib/widgets/menu_button.dart
> <https://github.com/mokuteki-io/dart-playgrounds/pull/5#discussion_r357750270>
> :
>
> > @@ -0,0 +1,18 @@
> +import 'package:flutter/material.dart';
> +
> +
> +class MenuButton extends StatelessWidget {
> +  const MenuButton({
> +    Key key,
> +  }) : super(key: key);
> +
> +  @override
> +  Widget build(BuildContext context) {
> +    return IconButton(
> +      icon: Icon(
> +        Icons.menu,
> +      ),
> +      onPressed: () => Scaffold.of(context).openDrawer(),
>
> @matejthetree <https://github.com/matejthetree> Pa kada maknem iz
> Scaffolda endDrawer:, MenuButton() ne otvara ladicu.
> Ovako radi:
> class _AboutPageState extends State<AboutPage> { @override Widget
> build(BuildContext context) { return Scaffold( endDrawer: CustomDrawer(),
> appBar: AppBar( title: Text('About Page'), ),
> A ovako ne radi:
> class _AboutPageState extends State<AboutPage> { @override Widget
> build(BuildContext context) { return Scaffold( appBar: AppBar(actions:
> <Widget>[ MenuButton(), ], title: Text('About Page'), ),
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mokuteki-io/dart-playgrounds/pull/5?email_source=notifications&email_token=ACMX5IWTP2F2M4UGFDXZBXLQYPAMPA5CNFSM4JZFP5ZKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCPFAWIY#discussion_r357750270>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACMX5IROYBOHLANAGS2OVYLQYPAMPANCNFSM4JZFP5ZA>
> .
>

matejthetree commented:
hoces reci da se button pokaze i bez MenuButtona?

On Fri, Dec 13, 2019 at 6:30 PM lea88pu <notifications@github.com> wrote:

> *@lea88pu* commented on this pull request.
> ------------------------------
>
> In lib/widgets/menu_button.dart
> <https://github.com/mokuteki-io/dart-playgrounds/pull/5#discussion_r357752742>
> :
>
> > @@ -0,0 +1,18 @@
> +import 'package:flutter/material.dart';
> +
> +
> +class MenuButton extends StatelessWidget {
> +  const MenuButton({
> +    Key key,
> +  }) : super(key: key);
> +
> +  @override
> +  Widget build(BuildContext context) {
> +    return IconButton(
> +      icon: Icon(
> +        Icons.menu,
> +      ),
> +      onPressed: () => Scaffold.of(context).openDrawer(),
>
> Ali zasto da ga stavim kad radi i bez njega?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mokuteki-io/dart-playgrounds/pull/5?email_source=notifications&email_token=ACMX5IVZJDDUQU7MRWZDBA3QYPBDFA5CNFSM4JZFP5ZKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCPFBOWY#discussion_r357752742>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACMX5IXWXXPYPGGJNVMKRZDQYPBDFANCNFSM4JZFP5ZA>
> .
>

lea88pu commented:
Da radi i bez njega

pet, 13. pro 2019. u 18:56 matejthetree <notifications@github.com> napisao
je:

> hoces reci da se button pokaze i bez MenuButtona?
>
> On Fri, Dec 13, 2019 at 6:30 PM lea88pu <notifications@github.com> wrote:
>
> > *@lea88pu* commented on this pull request.
> > ------------------------------
> >
> > In lib/widgets/menu_button.dart
> > <
> https://github.com/mokuteki-io/dart-playgrounds/pull/5#discussion_r357752742
> >
> > :
> >
> > > @@ -0,0 +1,18 @@
> > +import 'package:flutter/material.dart';
> > +
> > +
> > +class MenuButton extends StatelessWidget {
> > + const MenuButton({
> > + Key key,
> > + }) : super(key: key);
> > +
> > + @override
> > + Widget build(BuildContext context) {
> > + return IconButton(
> > + icon: Icon(
> > + Icons.menu,
> > + ),
> > + onPressed: () => Scaffold.of(context).openDrawer(),
> >
> > Ali zasto da ga stavim kad radi i bez njega?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/mokuteki-io/dart-playgrounds/pull/5?email_source=notifications&email_token=ACMX5IVZJDDUQU7MRWZDBA3QYPBDFA5CNFSM4JZFP5ZKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCPFBOWY#discussion_r357752742
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ACMX5IXWXXPYPGGJNVMKRZDQYPBDFANCNFSM4JZFP5ZA
> >
> > .
> >
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/mokuteki-io/dart-playgrounds/pull/5?email_source=notifications&email_token=AMXIZSYYRUFJLW6Q7WVKFV3QYPEDPA5CNFSM4JZFP5ZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEG2XXTA#issuecomment-565541836>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMXIZS6OXKOGL564OQP5GVDQYPEDPANCNFSM4JZFP5ZA>
> .
>

Commit:
Removed Material App from home and About page
Route changed,
Theme color set,
Text 'This app has been brought..'fits every screen,
MenuButton widget deleted.

djkowa commented:
@matejthetree Napravili smo promjene i pushali

matejthetree commented on lib/ui_pages/about_page.dart:
```diff
@@ -0,0 +1,108 @@
+  _AboutPageState createState() => _AboutPageState();
+}
+
+class _AboutPageState extends State<AboutPage> {
```
zbog cega vam treba statefull widget? :D

matejthetree commented on lib/ui_pages/home.dart:
```diff
@@ -0,0 +1,30 @@
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
```
ovdje isto ne vidim razlog za statefull

matejthetree commented on lib/widgets/drawer.dart:
```diff
@@ -0,0 +1,31 @@
+          Padding(
+            padding: const EdgeInsets.all(28.0),
+            child: FlatButton(
+              child: Text('About',style: TextStyle(fontSize: 20,fontStyle: FontStyle.italic) ),
```
mislim da ovdje fali formatting. dartfmt

matejthetree reviewed (changes requested):
super. samo jos detalji. odlican posao

Commit:
Stateful to Stateless Widget and formatted code
